### PR TITLE
feat(spring): 0.6.0 Spring unification track — unified starter, sovereign profile, TCK

### PR DESCRIPTION
**Pull request by GionaGranchelli**
# 0.6.0 Spring unification track → master (final integration)

The complete Spring unification track (S0–S7 + hardening H1–H3), synchronized with current master.

## Track history (all reviewed and merged into `track/0.6.0-spring-unification`)

- **S0–S5** (#281–#287) — sovereign runtime baseline, structured output, orchestration, security, standalone
- **S6** (#288, `70dd93f0`) — unified `tramai-spring-boot-starter`: renamed `tramai-spring-boot-starter-sovereign` → `tramai-spring-sovereign` (identical packages), deleted `SovereignDefaultProfileEnvironmentPostProcessor` + `spring.factories`, composition-only unified starter with 8 invariant tests, BOM/module-boundaries/catalog/CI/examples/docs updated, MQ-0020/21 deviations
- **S7** (#289, `04942322`) — Spring runtime profile compatibility TCK (contract + fixture + JUnit driver locking the two-profile contract), 5 docs files, `EnableTramai` KDoc
- **H1** (#292, `12e2f1a8`) — provider/authority unification: shared `SpringProviderResolution` seam, `tramai.profile` as sole runtime selector (`tramai.sovereign.enabled` gate removed), zero-provider deterministic startup failure, one-authority guard (incl. prototype-scoped beans), consumer smoke + persistence tests updated
- **H2** (#293, `ec73a1ef`) — `@AiTool` through Spring proxies: JDK-proxy discovery + interface-invocable dispatch preserving AOP advice, overload-safe exact signature resolution, cached invocable function
- **H3** (#294, `d0aea484`) — docs alignment: unified starter onboarding, executable-truth quickstarts (BOM placeholders, provider base-url, real persistence schema, fixed JDBC YAML), module-boundary cleanup

## Master sync (this PR)

Merge of `origin/master` (`f503721e`, Epic 8.2 state-machine property tests) into the track, commit `170ded39`. Expected overlap confined to two sovereign persistence test files (H1 provider-config wiring + master's outbox stub `expectedAttemptCount` parameter) — verified preserved after merge:
- `SovereignFilePersistenceAutoConfigurationTest.kt` +2 (master stub signature)
- `SovereignJdbcPersistenceAutoConfigurationTest.kt` +2 (master stub signature)
- H1's `MinimalProviderConfig`/`StubModelProvider`/`test-provider` wiring intact in both

## Integration-hardening commit (8aecc74b, after combined-tree review)

Final boundary-condition fixes from the combined-tree review, committed directly on the track:

1. **Symmetric authority guard (P1)** — `StandardProfileSovereignAuthorityGuardAutoConfiguration` in tramai-spring-sovereign: standard/missing profile + manual `SovereignTramai` bean → deterministic startup failure (one-authority invariant was asymmetric: sovereign rejected manual `Tramai`, but standard did not reject manual `SovereignTramai`). Prototype-scoped beans included.
2. **JDK-proxied `@AiTool` (P1)** — (a) implementation-only annotations no longer silently skipped (JDK proxies bypass the prefilter and are re-checked on the AOP target class); (b) generic interfaces resolve via type-parameter substitution (`Tool<T>` + `Tool<InvoiceInput>` impl). Both proven red pre-fix / green post-fix with AOP advice asserted.
3. **RC trigger coverage (P2)** — RC workflow now triggers on `tramai-spring-core/**`, `...-ops-actuator/**`, `...-ops-micrometer/**`; path set covers every `sovereignBundleModuleNames` module.
4. **KDoc cleanup** — `TramaiEnableImportSelector` no longer claims sovereign runtime creation depends on an application `ModelProvider` (obsolete since H1).

Gates re-run on the merged tree: spring-core/sovereign/starter tests ✅ · `verifyPr` ✅ · `verifyVersionAlignment` ✅ · `verifyMaintainabilityBaseline` ✅ · `verifySovereignRuntimeReleaseCandidate` ✅.

## Gate matrix (post-sync, local)

- `verifyPr` — **PASSED** (all subproject tests, build-logic tests, maintainability baseline, change policy: 88 files, runtime-behaviour, no violations)
- `verifySovereignRuntimeReleaseCandidate` — **PASSED**:
  - release manifest ✅ · API boundary ✅ · signed bundle ✅ · verification-repo closure ✅
  - consumer smoke ✅ · publication ✅ · document-intelligence evidence run ✅
- `verifyVersionAlignment` ✅ (from H3; re-run on merged tree)
- `verifyMaintainabilityBaseline` ✅

## Scope

- All Spring modules (spring-core, spring-sovereign, unified starter, sovereign add-on starters, provider adapters)
- Sovereign runtime core (sovereign, security, persistence, ops)
- Orchestration/outbox Epic 8.2 additions from master (state-machine property tests)
- Docs, module boundaries, BOM, CHANGELOG
- No 0.7/control-plane work, no redesign of S0–S6

## Review focus

- Two-profile contract locked by the S7 TCK: `tramai.profile` is the sole runtime selector; missing/standard → standard runtime, sovereign/SOVEREIGN → sovereign, invalid → deterministic startup failure
- One-authority invariant: sovereign + plain `Tramai` bean (incl. prototype) → startup failure; sovereign profile with zero resolved providers → startup failure
- Proxy-invoked `@AiTool` preserves Spring AOP advice and exact overload resolution
- Docs teach executable truth (startable minimal configs, correct persistence schema, version placeholders)

### CODE REVIEW

**Review by GionaGranchelli — COMMENTED** [review]
Final combined-tree review: DO NOT MERGE yet. The post-sync ancestry and gate matrix are clean, but I found three integration-level gaps that violate the contract or release-gate closure.

1. **Runtime authority guard is asymmetric (blocking).** H1 correctly rejects `tramai.profile=sovereign` when any plain `Tramai` bean exists, including prototypes. The reverse is not enforced: standard/missing profile still activates `TramaiAutoConfiguration`, whose `@ConditionalOnMissingBean` only considers `Tramai`; a user-supplied `SovereignTramai` is a distinct type, so standard can create a plain `Tramai` alongside it. That contradicts the public "exactly one runtime authority" guarantee. Keep spring-core sovereign-independent; add a guard from the sovereign integration for standard/missing profile that detects any manual `SovereignTramai` authority (include non-singletons, avoid eager init) and fails deterministically. Add explicit-standard and missing-profile regressions; prototype coverage is preferable.

2. **JDK-proxy `@AiTool` support still has uncovered correctness holes (blocking).** `AiToolScanner.findAnnotatedBeans` pre-filters using `factory.getType(beanName,false)` and that type's declared interfaces *before* it calls `AopUtils.getTargetClass(bean)`. For a JDK proxy where `@AiTool` is declared only on the implementation method (normal Spring usage) and the interface is unannotated, the proxy/interface prefilter sees no annotation and silently skips the bean. H2 tests annotate the interface, so they do not cover this path. Add an implementation-only annotation regression that proves discovery + invocation through the proxy + AOP advice.

   There is also a generic-interface form of the same proxy-resolution problem: the custom `sameSignatureAs` compares Kotlin type shapes. For `interface Tool<T> { @AiTool fun lookup(T) }` implemented as `Tool<InvoiceInput>`, the interface KFunction parameter classifier is the type parameter `T`, while the implementation method is `InvoiceInput`; the matcher cannot resolve the interface annotation/invocable function. That can either lose the tool or fall back to the target KFunction and reintroduce the JDK receiver mismatch. Add a directly generic-interface JDK-proxy regression. Prefer JVM/Spring invocable-method / bridge/generic resolution over expanding a bespoke KType matcher, while preserving the existing same-name/same-arity overload and suspend tests.

3. **Sovereign RC workflow trigger paths do not match the signed-bundle boundary (blocking release hygiene).** `sovereignBundleModuleNames` now includes `tramai-spring-core`, `...ops-actuator`, and `...ops-micrometer`, but `.github/workflows/sovereign-runtime-release-candidate.yml` does not watch those three paths. This is material now: spring-core contains H1 provider resolution and H2 tool scanning used by sovereign, and H2 itself demonstrated that a spring-core-only PR does not trigger RC. Add:
   - `tramai-spring-core/**`
   - `tramai-spring-boot-starter-sovereign-ops-actuator/**`
   - `tramai-spring-boot-starter-sovereign-ops-micrometer/**`
   Ideally add a repo-local guard tying the workflow coverage to the bundle module set so this cannot drift again.

Non-blocking cleanup while touching the branch:
- `TramaiEnableImportSelector` KDoc still says sovereign runtime creation is conditional on an application `ModelProvider`; H1 removed that condition. Update it.
- PR body says "zero-provider under a selected profile → startup failure"; narrow to sovereign profile, since the standard runtime is not governed by that H1 invariant.
- The current Copilot `lowercase()` locale thread is a false positive: modern Kotlin `String.lowercase()` is locale-invariant; no `Locale.ROOT` change is needed solely for that comment.

Verified good in this review: merge commit `170ded39` has exactly the H3 and current-master parents; track is 13 ahead/0 behind; both persistence overlap files retain H1 provider fixtures plus master's `expectedAttemptCount`; unified starter is composition-only; BOM/publication/signed-bundle include the new canonical Spring artifacts; old sovereign starter directory is gone; current CI/zero-egress/RC/maintainability/GitGuardian are green.

After the fixes, rerun targeted Spring tests + unified TCK, `verifyPr`, `verifySovereignRuntimeReleaseCandidate`, `verifyVersionAlignment`, `verifyMaintainabilityBaseline`, then require fresh PR CI/zero-egress/RC/maintainability/GitGuardian and zero unresolved review threads before master merge.

File: tramai-spring-core/src/main/kotlin/dev/tramai/spring/TramaiRuntimeProfileSupport.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    fun validate(configuredProfile: String?) {
+        if (configuredProfile == null) {
+            return
+        }
+        require(configuredProfile.lowercase() in supportedProfiles) {
+            "Unsupported tramai.profile '$configuredProfile'. Supported values: ${supportedProfiles.joinToString(", ")}."
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (GionaGranchelli) [reply tramai-spring-core/src/main/kotlin/dev/tramai/spring/TramaiRuntimeProfileSupport.kt:16]: Not a defect here. In current Kotlin, `String.lowercase()` (without a locale argument) performs locale-invariant Unicode case conversion; locale-specific behavior requires the explicit `lowercase(locale)` overload. The old default-locale-sensitive API was `toLowerCase()`. Keeping `lowercase()` is deterministic for profile parsing.
